lib/msal-core/src/Logger.ts

```diff
@@ -74,23 +75,23 @@ export class Logger {// Singleton Class
           piiLoggingEnabled = false
       } = options;
 
-      this._localCallback = localCallback;
-      this._correlationId = correlationId;
-      this._level = level;
-      this._piiLoggingEnabled = piiLoggingEnabled;
+      this.localCallback = localCallback;
+      this.correlationId = correlationId;
+      this.level = level;
+      this.piiLoggingEnabled = piiLoggingEnabled;
   }
 
   /**
    * @hidden
    */
   private logMessage(logLevel: LogLevel, logMessage: string, containsPii: boolean): void {
-    if ((logLevel > this._level) || (!this._piiLoggingEnabled && containsPii)) {
+    if ((logLevel > this.level) || (!this.piiLoggingEnabled && containsPii)) {
       return;
     }
-    var timestamp = new Date().toUTCString();
-    var log: string;
-    if (!Utils.isEmpty(this._correlationId)) {
-      log = timestamp + ":" + this._correlationId + "-" + Utils.getLibraryVersion() + "-" + LogLevel[logLevel] + " " + logMessage;
+    const timestamp = new Date().toUTCString();
+    let log: string;
```

Comment:
could factor this into a small helper 

const log: string = getFormattedLogMessage(this.correlationId, logMessage;

Reply:
Please diff between api_iter_2 and api_iter_3: This PR is meant to be one more increment focusing on one specific functionality. I am not addressing any other optimizations for this PR.

Reply:
@DarylThayil But lets also close loop on these by the end of the tunnel once we have main functionality working.